src/network-uptime/network-dashboard.py

```diff
@@ -0,0 +1,154 @@
+    #device_name_data = device_name_data.add_suffix('_count')
+    device_name_data.index = pd.to_datetime(device_name_data.index)
+    device_name_data['pct_uptime'] = (
+        device_name_data['uptime']/device_name_data.sum(axis=1)*100).round(1)
+    device_name_data['pct_downtime'] = 100 - device_name_data['pct_uptime']
+
+    return device_name_data
+
+
+json_data = load_device_data(selected_device)
+device_name_data = load_device_dataframe(json_data)
+
+device_name_data_year = device_name_data.index.year.unique()
+selected_year = st.sidebar.selectbox('Year', device_name_data_year)
+device_name_data_month_name = device_name_data[device_name_data.index.year ==
+                                               selected_year].index.month_name().unique()
+device_name_data_month = device_name_data[device_name_data.index.year ==
+                                          selected_year].index.month.unique()
+
+device_name_data_month_wrap = dict(
+    zip(device_name_data_month_name, device_name_data_month))
+selected_month = st.sidebar.radio('Month', device_name_data_month_name)
+
+
+# Barchart
+my_color = ['g', 'r']
+fig_bar, axes_bar = plt.subplots(1, 1, figsize=(14, 10))
+device_name_data_subset = device_name_data[(device_name_data.index.year == selected_year) & (
+    device_name_data.index.month == device_name_data_month_wrap[selected_month])][['uptime', 'downtime']]
+
+date_max_value = device_name_data_subset.index.days_in_month.unique().item()
+selected_date = st.sidebar.slider(
+    'day', min_value=1, max_value=date_max_value, value=date_max_value)
+
+device_name_data_subset = device_name_data_subset[(
+    device_name_data_subset.index.day >= 1) & (device_name_data_subset.index.day <= selected_date)]
+device_name_data_subset.index = device_name_data_subset.index.date
+
+
+device_name_data_subset.plot(kind='bar', ax=axes_bar, color=my_color)
+axes_bar.yaxis.set_major_locator(MaxNLocator(integer=True))
+axes_bar.set_title(
+    f'{selected_device} device network uptime and downtime daily count for the month of {selected_month}, {selected_year}', fontsize=20)
+plt.tight_layout(pad=5)
+
+# Piechart
+
+device_name_data_months_group = device_name_data.groupby(
+    pd.Grouper(freq='M')).sum()[['uptime', 'downtime']]
+device_name_data_months_pct = device_name_data_months_group.div(
+    device_name_data_months_group.sum(axis=1), axis='rows').round(2)
+fig_pie, axes_pie = plt.subplots(1, 1, figsize=(10, 8))
+axes_pie.set_title(
+    f'{selected_device} device network uptime and downtime percent for the month of {selected_month}, {selected_year}', fontsize=15)
+device_name_data_month_pct = device_name_data_months_pct[(device_name_data_months_pct.index.year == selected_year) & (
+    device_name_data_months_pct.index.month == device_name_data_month_wrap[selected_month])]
+
+device_name_data_month_pct.T.plot.pie(
+    y=device_name_data_month_pct.index[0], colors=['green', 'red'], autopct='%0.1f%%', ax=axes_pie)
+axes_pie.yaxis.set_visible(False)
```

Comment:
These lines plot the device piechart using the percentage uptime and downtime for the selected month